File: StudioSAT-V2/native/android/app/src/main/java/br/com/studiosat/v2/Station.kt
package br.com.studiosat.v2

/** Catálogo nativo. Nenhuma estação contém lógica de reprodução. */
data class Station(val id: String, val name: String, val streamUrl: String)

object Stations {
    val all = listOf(
        Station("radioprincipal", "Rádio Studio Sat", "https://radio.studiosatweb.com.br/radioprincipal/index.m3u8"),
        Station("radiopop", "Studio Sat Pop", "https://radio.studiosatweb.com.br/radiopop/index.m3u8"),
        Station("radiorock", "Studio Sat Rock", "https://radio.studiosatweb.com.br/radiorock/index.m3u8"),
        Station("radioclassicas", "Studio Sat Clássicas", "https://radio.studiosatweb.com.br/radioclassicas/index.m3u8"),
        Station("radiocountry", "Studio Sat Country", "https://radio.studiosatweb.com.br/radiocountry/index.m3u8")
    )
}

File: StudioSAT-V2/native/ios/StudioSatV2/Station.swift
import Foundation

/// Modelo puro. Não contém qualquer processamento de áudio.
struct Station: Identifiable, Hashable {
    let id: String
    let name: String
    let hlsURL: URL

    var rawAACURL: URL {
        URL(string: "https://www.radio.studiosatweb.com.br/listen-v2/live/\(id).aac")!
    }

    static let all: [Station] = [
        .init(id: "radioprincipal", name: "Rádio Studio Sat", hlsURL: URL(string: "https://radio.studiosatweb.com.br/radioprincipal/index.m3u8")!),
        .init(id: "radiopop", name: "Studio Sat Pop", hlsURL: URL(string: "https://radio.studiosatweb.com.br/radiopop/index.m3u8")!),
        .init(id: "radiorock", name: "Studio Sat Rock", hlsURL: URL(string: "https://radio.studiosatweb.com.br/radiorock/index.m3u8")!),
        .init(id: "radioclassicas", name: "Studio Sat Clássicas", hlsURL: URL(string: "https://radio.studiosatweb.com.br/radioclassicas/index.m3u8")!),
        .init(id: "radiocountry", name: "Studio Sat Country", hlsURL: URL(string: "https://radio.studiosatweb.com.br/radiocountry/index.m3u8")!)
    ]
}
